### ISSUE TITLE
Liquibase-CDI fails to deploy after successful run

### ISSUE THREAD

**Issue by gmussi** [payara/Payara#2834]
# Description #
----------

When deploying Liquibase-CDI to Payara Micro (5.181 and 5.182) the deploy fails after it has been successfully run. This only happens in my project. I have re-implemented the same thing in a sample project and there it works. 
The error message does not give me enough to know what might be wrong in my project.

## Expected Outcome

The deploy should not fail after a successful run.

## Current Outcome

Currently, the following exception is being thrown:

```
16:05:10.263 [main] DEBUG liquibase.executor.jvm.JdbcExecutor - Release Database Lock
16:05:10.265 [main] DEBUG liquibase.executor.jvm.JdbcExecutor - UPDATE PUBLIC.DATABASECHANGELOGLOCK SET LOCKED = FALSE, LOCKEDBY = NULL, LOCKGRANTED = NULL WHERE ID = 1
16:05:10.266 [main] INFO liquibase.lockservice.StandardLockService - Successfully released change log lock
2018-06-12T16:05:10.286+0200] [] [SEVERE] [NCLS-CORE-00026] [javax.enterprise.system.core] [tid: _ThreadID=1 _ThreadName=main] [timeMillis: 1528812310286] [levelValue: 1000] [[
  Exception during lifecycle processing
org.glassfish.deployment.common.DeploymentException: CDI deployment failure:WELD-000049: Unable to invoke public void liquibase.integration.cdi.CDILiquibase.onStartup() on liquibase.integration.cdi.CDILiquibase@5f0b2fe1 -- WELD-000049: Unable to invoke public void liquibase.integration.cdi.CDILiquibase.onStartup() on liquibase.integration.cdi.CDILiquibase@5f0b2fe1
    at org.jboss.weld.bootstrap.events.AbstractDeploymentContainerEvent.fire(AbstractDeploymentContainerEvent.java:38)
    at org.jboss.weld.bootstrap.events.AfterDeploymentValidationImpl.fire(AfterDeploymentValidationImpl.java:28)
    at org.jboss.weld.bootstrap.WeldStartup.validateBeans(WeldStartup.java:499)
    at org.jboss.weld.bootstrap.WeldBootstrap.validateBeans(WeldBootstrap.java:93)
    at org.glassfish.weld.WeldDeployer.processApplicationLoaded(WeldDeployer.java:511)
    at org.glassfish.weld.WeldDeployer.event(WeldDeployer.java:422)
    at org.glassfish.kernel.event.EventsImpl.send(EventsImpl.java:131)
    at org.glassfish.internal.data.ApplicationInfo.load(ApplicationInfo.java:333)
    at com.sun.enterprise.v3.server.ApplicationLifecycle.prepare(ApplicationLifecycle.java:496)
    at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:540)
    at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:549)
    at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:545)
    at java.security.AccessController.doPrivileged(Native Method)
    at javax.security.auth.Subject.doAs(Subject.java:360)
    at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:544)
    at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:575)
    at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:567)
    at java.security.AccessController.doPrivileged(Native Method)
    at javax.security.auth.Subject.doAs(Subject.java:360)
    at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:566)
    at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1475)
    at com.sun.enterprise.v3.admin.CommandRunnerImpl.access$1300(CommandRunnerImpl.java:111)
    at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1857)
    at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1733)
    at com.sun.enterprise.admin.cli.embeddable.DeployerImpl.deploy(DeployerImpl.java:133)
    at fish.payara.micro.impl.PayaraMicroImpl.deployAll(PayaraMicroImpl.java:1553)
    at fish.payara.micro.impl.PayaraMicroImpl.bootStrap(PayaraMicroImpl.java:1036)
    at fish.payara.micro.impl.PayaraMicroImpl.main(PayaraMicroImpl.java:199)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at fish.payara.micro.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
    at fish.payara.micro.boot.loader.Launcher.launch(Launcher.java:107)
    at fish.payara.micro.boot.loader.Launcher.launch(Launcher.java:70)
    at fish.payara.micro.boot.PayaraMicroLauncher.main(PayaraMicroLauncher.java:79)
    at fish.payara.micro.PayaraMicro.main(PayaraMicro.java:397)
]]

[2018-06-12T16:05:10.287+0200] [] [SEVERE] [] [javax.enterprise.system.core] [tid: _ThreadID=1 _ThreadName=main] [timeMillis: 1528812310287] [levelValue: 1000] Exception while loading the app

[2018-06-12T16:05:10.293+0200] [] [SEVERE] [AS-WEB-GLUE-00192] [javax.enterprise.web] [tid: _ThreadID=1 _ThreadName=main] [timeMillis: 1528812310293] [levelValue: 1000] Undeployment failed for context
```

## Steps to reproduce (Only for bug reports) 

To reproduce, I simply followed the liquibase tutorial:
http://www.liquibase.org/documentation/cdi.html

### Samples

https://github.com/gmussi/liquibase-payara-micro-example
I made this project from zero, and here it works. I simply copied and pasted from my other project.

## Environment ##

- **Payara Version**: 5.182
- **Edition**: Micro
- **JDK Version**: 8 Oracle
- **Operating System**: Mac
- **Database**: PostGres

**Comment by smillidge** [comment]
I suspect an exception is being thrown in onStartup

**Comment by gmussi** [comment]
Any idea how I can debug this? I supplied a logging file with everything possible enabled, but nothing useful came out. I inspected the Weld code and there is indeed no logging in this area. So I believe this exception is being suppressed.

**Comment by smillidge** [comment]
I would use a debugger and put a breakpoint on where the exception is thrown.

**Comment by mydeadlyvenoms** [comment]
@gmussi I have exactly the same issue. I also tried to use the `listener-class` via web.xml  e.g.,

```
<listener>
  <listener-class>liquibase.integration.servlet.LiquibaseServletListener</listener-class>
</listener>
```

But this doesn't create any tables, does this work for you?

**Comment by mydeadlyvenoms** [comment]
The strange thing is, it worked before I migrated from an EAR to a WAR project / file.

**Comment by gmussi** [comment]
I tested now deploying the same .war file in a full Payara 5.182 and it works there without problems.

The same .war file, when deploying using `java -jar payara-micro-5.182.jar --deploy ROOT.war` throws this Exception.
I am not really sure how I can debug the application with Payara Micro. I will try now to debug in eclipse by Instantiating Payara Micro via Java code.

**Comment by smillidge** [comment]
You can also debug Payara Micro using the standard command line options and debugging remotely `-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=1044`

**Comment by mydeadlyvenoms** [comment]
Hi everyone, sorry I made a mistake (some maven issues)... using the `liquibase.integration.servlet.LiquibaseServletListener` via `web.xml` works.

**Comment by gmussi** [comment]
hi @smillidge ,

You were right that an exception was being thrown. Using the remote debug feature (never had used it before, hehe) I was able to dig around and find the exception message.

Moving from 5.181 to 5.182 one of the points I saw you guys working on, was to improve exception logging and eat less exceptions. This is much improved with 5.182. I would say that there is still some way to go here. :)

Thanks for the tips!
